File: plugin/src/test/java/io/github/pshevche/act/ActPluginTest.java
package io.github.pshevche.act;

import org.gradle.testfixtures.ProjectBuilder;
import org.gradle.api.Project;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
 * A simple unit test for the 'io.github.pshevche.greeting' plugin.
 */
class ActPluginTest {
    @Test void pluginRegistersATask() {
        // Create a test project and apply the plugin
        Project project = ProjectBuilder.builder().build();
        project.getPlugins().apply("io.github.pshevche.greeting");

        // Verify the result
        assertNotNull(project.getTasks().findByName("greeting"));
    }
}

File: plugin/src/test/kotlin/io/github/pshevche/act/ActPluginTest.kt
package io.github.pshevche.act

import io.kotest.core.spec.style.FreeSpec
import io.kotest.matchers.shouldNotBe
import org.gradle.testfixtures.ProjectBuilder

class ActPluginTest : FreeSpec({

    "plugin registers task" {
        val project = ProjectBuilder.builder().build()
        project.plugins.apply("io.github.pshevche.act")
        project.tasks.findByName("greeting") shouldNotBe null
    }
})
